src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcApplicationBuilderExtensions.cs

```diff
@@ -96,5 +98,53 @@ public static IApplicationBuilder UseMvcWithDefaultRoute(this IApplicationBuilde
 
             return app.UseRouter(routes.Build());
         }
+
+        public static IApplicationBuilder UseMvcWithMiddleware(
```

Comment:
I don't want to get too distracted by the way this works, but the the problem we need to solve is generally going to be related to configuring different middleware for different actions. This only gives us a single pipeline, but at least the lifetime is right.